### ISSUE TITLE
remove usage and references to twisted.python.plugin

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1911]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#1911|
|Type|task|
|Created|2006-07-11 13:44:41Z|
|Branch|https://github.com/twisted/twisted/tree/remove-old-plugins-1911|

It's been a while since that module was deprecated. We can probably drop support for it now.

mktap still uses it, for example.

We should also check if any documentation still refers to it.

<details><summary>Searchable metadata</summary>

```
trac-id__1911 1911
type__task task
reporter__radix radix
priority__high high
milestone__ 
branch__branches_remove_old_plugins_1911 branches/remove-old-plugins-1911
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1152625481000000 1152625481000000
changetime__1231177275000000 1231177275000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

mktap is gone, can we delete this now?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra removed owner|
|-|-|

```
$ grep -R 'twisted.python.plugin' .
./doc/historic/2003/pycon/applications/applications.html:    <p>The <code>twisted.python.plugin</code> provides the most high-level
./twisted/python/plugin.py:        warnings.warn("The twisted.python.plugin system is deprecated.  "
./twisted/python/plugin.py:    warnings.warn("The twisted.python.plugin system is deprecated.  "
./twisted/topfiles/ChangeLog.Old:	and scandir(3) for use by twisted.python.plugins.
./twisted/topfiles/ChangeLog.Old:	* twisted/python/plugin.py: fixed a bug that got the wrong plugins.tml
```

Only the historic documentation still has some references to t.p.plugin, which should stay that way I guess. So I think it's safe to get rid of this module?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Some code still uses it in order to load old-style plugins:

```
exarkun@boson:~/Projects/Twisted/trunk$ grep twisted.python twisted/ -r --include '*.py' | grep import | grep plugin
twisted/scripts/mktap.py:from twisted.python import usage, util, plugin as oldplugin
twisted/lore/scripts/lore.py:from twisted.python import usage, plugin as oldplugin, reflect
exarkun@boson:~/Projects/Twisted/trunk$ 
```

Fortunately that's not much.  It will need to be deleted (along with any tests, gosh I hope there are some) when `twisted.python.plugin` is removed.

For reference, `twisted.python.plugin` was deprecated 2005-03-13.  The first release to include the deprecation was Twisted 2.0.0.  I think this indeed satisfies our compatibility requirements and the module can be removed.  We may as well wait until 8.2 is done, though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [25763]) Branching to 'remove-old-plugins-1911'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to @exarkun|
|-|-|

The changes looks good, but some tests for the changes in mktap and lore scripts would be welcome.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

(In [25770]) simple test for loadPlugins

refs [#1911](https://github.com/twisted/twisted/issues/1911)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun removed owner|
|-|-|

Added a couple tests, in r25770 and r25771.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set owner to @exarkun|
|-|-|

Thanks, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set status to `closed`|
|-|-|

(In [25788]) Merge remove-old-plugins-1911

Author: exarkun
Reviewer: therve
Fixes: [#1911](https://github.com/twisted/twisted/issues/1911)

Remove the old Twisted plugin API and adjust the two remaining users of it so that
they only use the new Twisted plugin API.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
